src/operator/warp-inl.h

```diff
@@ -0,0 +1,151 @@
+/*!
+ * Copyright (c) 2016 by Contributors
+ * \file warp-inl.h
+ * \brief warp operator and symbol
+ * \author Xu Dong 
+*/
+#ifndef MXNET_OPERATOR_WARP_INL_H_
+#define MXNET_OPERATOR_WARP_INL_H_
+#include <dmlc/logging.h>
+#include <dmlc/parameter.h>
+#include <mxnet/operator.h>
+#include <map>
+#include <vector>
+#include <string>
+#include <utility>
+#include "./mshadow_op.h"
+#include "./operator_common.h"
+namespace mxnet {
+namespace op {
+//  Declare enumeration of input order to make code more intuitive.
+//  These enums are only visible within this header
+namespace Warp {
+enum  WarpOpInputs{kData, kGrid};
+enum  WarpOpOutputs{kOut};
+}  // namespace Warp
+
+struct WarpParam : public dmlc::Parameter<WarpParam> {
+  bool only_grid;
+  DMLC_DECLARE_PARAMETER(WarpParam) {
+  DMLC_DECLARE_FIELD(only_grid).set_default(false)
+  .describe("indicate whether WarpOp only calculate the gradient of the grid ");

```

Comment:
We should be able to know `only_grid` option by checking req in the backward pass